### PR TITLE
Minor changes in Hunting queries

### PR DESCRIPTION
**Pull request by vakohl**
Required items, please complete.
   
   Change(s):
   - Various minor changes

   Reason for Change(s):
   - Query validations

   Version Updated:
   - various versions

   Testing Completed:
   - Basic testing done with the available data.

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Review by shainw — DISMISSED** [review]
Some minor changes.

File: Solutions/Azure Activity/Hunting Queries/Anomalous_Listing_Of_Storage_Keys.yaml
[the remark this answers is not in the document]
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/Anomalous_Listing_Of_Storage_Keys.yaml:1]: thanks @shainw . done

File: Solutions/Azure Activity/Hunting Queries/AzureRunCommandFromAzureIP.yaml
[the remark this answers is not in the document]
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/AzureRunCommandFromAzureIP.yaml:1]: thanks @shainw done.

File: Solutions/Azure Activity/Hunting Queries/AzureSentinelConnectors_AdministrativeOperations.yaml
[the remark this answers is not in the document]
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/AzureSentinelConnectors_AdministrativeOperations.yaml:1]: done.

File: Solutions/Azure Activity/Hunting Queries/Granting_Permissions_to_Account.yaml
[the remark this answers is not in the document]
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/Granting_Permissions_to_Account.yaml:1]: done.

File: Solutions/Azure Activity/Hunting Queries/AzureNSG_AdministrativeOperations.yaml [hunk 1]
```diff
@@ -12,23 +12,23 @@ tactics:
 relevantTechniques:
   - T1496
 query: | 
- 
   let opValues = dynamic(["Microsoft.Network/networkSecurityGroups/write", "Microsoft.Network/networkSecurityGroups/delete"]);
   // Azure NSG Create / Update / Delete
   AzureActivity
   | where Category == "Administrative"
```
Comment (devikamehra) [inline Solutions/Azure Activity/Hunting Queries/AzureNSG_AdministrativeOperations.yaml:18]: Change == to =~

File: Solutions/Azure Activity/Hunting Queries/AzureSentinelWorkbooks_AdministrativeOperation.yaml [hunk 2]
```diff
@@ -12,23 +12,23 @@ tactics:
 relevantTechniques:
   - T1496
 query: | 
- 
   let opValues = dynamic(["microsoft.insights/workbooks/write", "microsoft.insights/workbooks/delete"]);
   // Microsoft Sentinel Workbook Create / Update / Delete
   AzureActivity
   | where Category == "Administrative"
```
Comment (devikamehra) [inline Solutions/Azure Activity/Hunting Queries/AzureSentinelWorkbooks_AdministrativeOperation.yaml:18]: == to =~

File: Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml [hunk 3]
```diff
@@ -14,33 +14,34 @@ relevantTechniques:
...
 query: |
-
+  let lookback = 7d;
   AzureActivity
+  | where TimeGenerated >= ago(lookback)
   | where OperationNameValue has_any ("ipfilterrules", "securityRules", "publicIPAddresses", "firewallrules") and OperationNameValue endswith "write"
   // Choosing Accepted here because it has the Rule Attributes included
   | where ActivityStatusValue == "Accepted" 
   // If there is publicIP info, include it
-  | extend publicIPAddress_ = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).ipAddress)
```
Comment (devikamehra) [inline Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml:23]: We are removing publicIPAddress_, do we not need it?

File: Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml [hunk 4]
```diff
@@ -14,33 +14,34 @@ relevantTechniques:
...
   AzureActivity
+  | where TimeGenerated >= ago(lookback)
   | where OperationNameValue has_any ("ipfilterrules", "securityRules", "publicIPAddresses", "firewallrules") and OperationNameValue endswith "write"
   // Choosing Accepted here because it has the Rule Attributes included
   | where ActivityStatusValue == "Accepted" 
   // If there is publicIP info, include it
-  | extend publicIPAddress_ = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).ipAddress) 
-  | extend publicIPAddressVersion_ = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).publicIPAddressVersion) 
-  | extend publicIPAllocationMethod_ = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).publicIPAllocationMethod) 
+  | extend publicIPAddressVersion = tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties.publicIPAddressVersion)
```
Comment (devikamehra) [inline Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml:24]: Same value of responseBody is getting parsed multiple times. Can we do this once and use it in different locations?

File: Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml [hunk 5]
```diff
@@ -14,33 +14,34 @@ relevantTechniques:
...
   | extend destinationPortRange = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).destinationPortRange) 
   | extend direction = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).direction) 
   | extend protocol = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).protocol) 
   | extend sourcePortRange = tostring(parse_json(tostring(parse_json(tostring(parse_json(Properties).responseBody)).properties)).sourcePortRange) 
-  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), ResourceIds = makeset(ResourceId) by Caller, CallerIpAddress, Resource, ResourceGroup, 
-  ActivityStatusValue, ActivitySubstatus, SubscriptionId, access, description, destinationPortRange, direction, protocol, sourcePortRange  
-  | extend timestamp = StartTime, AccountCustomEntity = Caller, IPCustomEntity = CallerIpAddress
-
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), ResourceIds = makeset(_ResourceId) by Caller, CallerIpAddress, Resource, ResourceGroup,
```
Comment (devikamehra) [inline Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml:33]: Make_set is missing the max limit

File: Solutions/Azure Activity/Hunting Queries/Rare_Custom_Script_Extension.yaml [hunk 6]
```diff
@@ -35,14 +34,14 @@ query: |
   let RareCustomScriptExecution =  CurrentCustomScriptExecution
```
Comment (devikamehra) [inline Solutions/Azure Activity/Hunting Queries/Rare_Custom_Script_Extension.yaml:34]: Value of properties is getting parsed multiple times.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml (hunk 3 above)
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml:23]: I did not find this column value in the sample data I used. Also, this column along with other columns extracted from JSON were being calculated but was not being used. I added usage under summarize for the other columns for which I found values.

File: Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml (hunk 4 above)
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml:24]: Thanks @devikamehra . This is done

File: Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml (hunk 5 above)
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/PortOpenedForAzureResource.yaml:33]: done

File: Solutions/Azure Activity/Hunting Queries/AzureSentinelWorkbooks_AdministrativeOperation.yaml (hunk 2 above)
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/AzureSentinelWorkbooks_AdministrativeOperation.yaml]: done

File: Solutions/Azure Activity/Hunting Queries/AzureNSG_AdministrativeOperations.yaml (hunk 1 above)
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/AzureNSG_AdministrativeOperations.yaml]: done

File: Solutions/Azure Activity/Hunting Queries/Rare_Custom_Script_Extension.yaml (hunk 6 above)
Comment (vakohl) [reply Solutions/Azure Activity/Hunting Queries/Rare_Custom_Script_Extension.yaml:35]: thanks @devikamehra . Parsing and storing Properties in a variable now.

**Review by devikamehra — APPROVED** [review]

**Review by anki-narravula — APPROVED** [review]

**Pull request merged**